src/WifiManager.cpp

```diff
@@ -133,14 +133,22 @@ void WifiManager::setupWiFi(void (*manageDisconnections)(), void (*manageHardwar
 }
 
 void WifiManager::reconnectToWiFi(void (*manageDisconnections)(), void (*manageHardwareButton)()) {
+  if(WiFi.status() == WL_CONNECTED){
+    wifiReconnectAttemp = 0;
+  }
 
-  wifiReconnectAttemp = 0;
-
+#if (NON_BLOCKING_RECONNECT)
+  static unsigned int lastReconnectAttempt = 0;
+  if(WiFi.status() != WL_CONNECTED && millis() - lastReconnectAttempt > DELAY_500){
+    manageHardwareButton();
+    lastReconnectAttempt = millis();
+#else
   // loop here until connection
   while (WiFi.status() != WL_CONNECTED) {
-
     manageHardwareButton();
```

Comment:
It would have been cleaner to put `manageHardwareButton()` after the `#if` block, but that would have meant that delay would happen before `manageHardwareButton()`. I don't know if this would have any unintended consequences so I opted for keeping it the way it was in exchange for slightly messier code.